policies/push-global.rego

```diff
@@ -1,75 +1,80 @@
+affected_files := input.push.affected_files
 
-# If pre-commit hooks make changes, they are not semantic changes
-# and can and should be ignored.
-ignore  { input.push.message == "pre-commit fixes" }
+# Track these extensions in the project folder
+tracked_extensions := {".tf", ".tf.json", ".tfvars", ".yaml"}
 
-# Fetch all of our affected files
-filepath := input.push.affected_files
+project_root := input.stack.project_root
 
-# Check if any Terraform files were modified in a project
-tf_affected {
-    startswith(filepath[_], input.stack.project_root)
-    endswith(filepath[_], ".tf")
+# Check if any of the tracked extensions were modified in the project folder
+# https://www.openpolicyagent.org/docs/latest/policy-language/#some-keyword
+# https://www.openpolicyagent.org/docs/latest/policy-language/#variable-keys
+# https://www.openpolicyagent.org/docs/latest/policy-reference/#iteration
+project_affected {
+    some i, j
+    startswith(affected_files[i], project_root)
+    endswith(affected_files[i], tracked_extensions[j])
 }
 
-# Check if any Terraform json files were modified in a project
-tf_affected {
-    startswith(filepath[_], input.stack.project_root)
-    endswith(filepath[_], ".tf.json")
-}
+# Split the stack name into a list
+stack_name_parts := split(input.stack.name, "-")
 
-# Check if any .tfvars files were modified in a project
-tf_affected {
-    startswith(filepath[_], input.stack.project_root)
-    endswith(filepath[_], ".tfvars")
+# Check if the environment has been modified
+stack_config_affected {
+    contains(affected_files[_], concat("-", [stack_name_parts[0], stack_name_parts[1]]))
 }
 
-# Check if any .yaml files were modified in a project
-tf_affected {
-    startswith(filepath[_], input.stack.project_root)
-    endswith(filepath[_], ".yaml")
-}
+# Get labels
+labels := input.stack.labels
 
-# Split our stack name into a list for matching below
-stack_name := split(input.stack.name, "-")
+# Get imports from the provided `labels`
+# https://www.openpolicyagent.org/docs/latest/policy-language/#comprehensions
+imports := [imp | startswith(labels[i], "import:"); imp := split(labels[i], ":")[1]]
 
-# Check if our global settings have been modified
-config_affected {
-    contains(filepath[_], "/globals.yaml")
+# Check if any of the imports have been modified
+stack_config_affected {
+    endswith(affected_files[_], imports[_])
 }
 
-# Check if our environment globals have been modified
-config_affected {
-    contains(filepath[_], concat("-", [stack_name[0], "globals"]))
-}
+# Get component stack dependencies from the provided `labels`
+stack_deps := [dep | startswith(labels[i], "stack-deps:"); dep := split(labels[i], ":")[1]]
 
-# Check if our environment has been modified
-config_affected {
-    contains(filepath[_], concat("-", [stack_name[0], stack_name[1]]))
+# Check if any of the stack dependencies have been modified
+stack_config_affected {
+    endswith(affected_files[_], stack_deps[_])
```

Comment:
```suggestion
stack_config_affected {
    endswith(affected_files[_], sprintf("/%v", [stack_deps[_]]))
```

Reply:
@aknysh Are you saying `stack_deps` also has `stacks/` in it? I cannot trace where `stack_deps` is coming from .

Reply:
added `stacks/` prefix to `stack_deps`